nixos/modules/services/network-filesystems/ipfs-cluster.nix

```
@@ -0,0 +1,133 @@
+{ config, lib, pkgs, options, ... }:
+with lib;
+let
+  cfg = config.services.ipfs-cluster;
+  opt = options.services.ipfs-cluster;
+
+  # secret is by envvar, not flag
+  initFlags = toString [
+    (optionalString (cfg.initPeers != [ ]) "--peers")
+    (lib.strings.concatStringsSep "," cfg.initPeers)
+  ];
+in {
+
+  ###### interface
+
+  options = {
+
+    services.ipfs-cluster = {
+
+      enable = mkEnableOption
+        "Pinset orchestration for IPFS - requires ipfs daemon to be useful";
+
+      user = mkOption {
+        type = types.str;
+        default = "ipfs";
+        description = "User under which the ipfs-cluster daemon runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "ipfs";
+        description = "Group under which the ipfs-cluster daemon runs.";
+      };
+
+      consensus = mkOption {
+        type = types.enum [ "raft" "crdt" ];
+        description = "Consensus protocol - 'raft' or 'crdt'. https://cluster.ipfs.io/documentation/guides/consensus/";
+      };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/ipfs-cluster";
+        description = "The data dir for ipfs-cluster.";
+      };
+
+      initPeers = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = "Peer addresses to initialize with on first run.";
+      };
+
+      openSwarmPort = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open swarm port, secured by the cluster secret. This does not expose the API or proxy. https://cluster.ipfs.io/documentation/guides/security/";
+      };
+
+      secretFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          File containing the cluster secret in the format of EnvironmentFile as described by
+          <citerefentry><refentrytitle>systemd.exec</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry>. For example:
+          <programlisting>
+          CLUSTER_SECRET=<replaceable>...</replaceable>
+          </programlisting>
+
+          if null, a new secret will be generated on first run.
+          A secret in the correct format can also be generated by: openssl rand -hex 32
+        '';
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    assertions = [
+      { assertion = cfg.enable -> config.services.kubo.enable;
+        message = "ipfs-cluster requires ipfs - configure and enable services.kubo";
+      }
```

```suggestion
      { 
        assertion = cfg.enable -> config.services.kubo.enable;
        message = "ipfs-cluster requires ipfs - configure and enable services.kubo";
      }
```

Minor nit that I think brings this in line with the styleguide